### PR TITLE
Migrate to mui 5

### LINKED ISSUE #707: Upgrade material-ui to MUI 5
The package got renamed from `@material-ui` to `@mui` and comes with many cool new components. It will definitely be worth it to upgrade but we need to of course make sure that it won't break anything since it's a major release and they even changed the package name.

Upgrade guide: https://mui.com/guides/migration-v4/

### PR DESCRIPTION
**Pull request by positiveimpact**
## What and Why
Closes #707 
We use Material UI for many of our components. We were still using the outdated version 4 even though mui 5 has been released in September 2021. This started putting real roadblocks into development which is why I finally went through the work to upgrade to mui 5. I followed the four steps in [this guide](https://mui.com/material-ui/migration/migration-v4/) and did lots of manual testing to ensure there are no unexpected sideeffects.

This PR has changes in 293 files but most of them are trivial. **Please go commit by commit when reviewing this!**

##  Changes

_Step 1 and 2 folllow the "getting started" section in the migration guide._

### [Step 1: simple refactors to imports, props and other minor things](https://github.com/climateconnect/climateconnect/commit/6d3b70b2c1364c94b36750820a22929c0f30b298)
Here I simply ran the codemods [preset-safe](https://mui.com/material-ui/migration/migration-v4/#preset-safe) which makes all the trivial important refactors.
These are all automatic changes and do not need to be reviewed.

### [Step 2: Keep underline="hover" as a standard](https://github.com/climateconnect/climateconnect/commit/4ea78529b3f9f4e6c2352993b1d8277fdb0ec2a3)
Here I simply ran the [underline="hover"](https://mui.com/material-ui/migration/migration-v4/#link-underline-hover) codemod which makes sure links visually behave the same way as before the upgrade.
These are all automatic changes and do not need to be reviewed.

_Step 3 folllows the "styles and themes" section in the migration guide._

### [Step 3: theme updates, remove thick/thin borders from theme](https://github.com/climateconnect/climateconnect/commit/21fd9f3b09389040e6728b6b75a9a78ddbd716fa)
Following the "styles and themes" section of the guide I made the following changes:
- Wrap `<StyledEngineProvider injectFirst>` around what is returned by _app.js as described in the guide
- - Add module augmentation for DefaultTheme in theme.ts
- Please ignore the wrapped `<StyledEngineProvider injectFirst>` and DefaultTheme augmentation in FashionDescription, FoodDescription and LayoutWrapper. They are removed in the `Cleanup step 1` commit and can therefore be ignored.
- Remove `px` suffix in css where it was used
- Refactor imports from `material-ui` to `mui` where it was not yet done 
- With mui 5 the breakpoints have changed and for the same behavior each breakpoint needs to be upgraded by one (e.g. "xs" -> "sm"). I have made this change by running a codemod provided by mui.
- Remove `thick` and `thin` borders from the theme because they weren't used in enough places. Instead I simply applied the css directly in the component
- Refactor css -> sx when declaring css in a component
- Adapt overrides to new structure of theme.ts

### [Step 4: update packages](https://github.com/climateconnect/climateconnect/pull/1223/commits/5dde891dc5369621dfe0d55e4e0246ea7033dbd2)
Here I updated all material UI packages to mui 5 and added required packages.

_Steps 5 through 8 follow the "Breaking changes to core components" section in the migration guide_

### [Step 5: Update Date Picker](https://github.com/climateconnect/climateconnect/pull/1223/commits/1a7c16de145448da273b5a96e13d789f74e201c1)
The Material UI Datepicker which we used in our `DatePicker` component was a seperate package compatible with material ui 4 which is no longer maintained. I followed the appropriate [guide](https://mui.com/x/react-date-pickers/getting-started/) to update it. You can test whether it still works by navigating to the third step of "Share project" and picking a date.

### [Step 6: update Autocomplete component](https://github.com/climateconnect/climateconnect/pull/1223/commits/2ad45dc3a50ef87fe9dc36fb2a0fb23490deec16)
As described in the [component migration guide](https://mui.com/material-ui/migration/v5-component-changes/#autocomplete) the `renderOption` function of Autocomplete now takes two arguments `(props, options`and returns an actual DOM element rather than `<React.Fragment>`. Also the `select-option` value of the `reason` argument in `handleChange(value, reason)`has been renamed to `selectOption`. I made these changes to all instances of the Autocomplete component. You can test this by searching for and selecting a location in the filters on the browse page.

### [Step 7: Fix buttons](https://github.com/climateconnect/climateconnect/pull/1223/commits/6bcf5ed017b1c3a76f3b990594bdfdd3cbc4c76f)
There were quite a few changes to the `Button` component which I addressed in this step. Some additional changes are made in Step 8.
- Fixed the feedback and tutorial button which were broken after the update
- In mui 5 all buttons the default grey color has been removed and all buttons have color="primary" by default now. To keep the old colors I applied the changes from the [CodeSandbox demo](https://codesandbox.io/s/mimic-v4-button-default-color-bklx8?file=/src/Demo.tsx) provided in the migration guide. Following the guide I introduced a "grey" color to the button component and applied it on all contained buttons that didn't have a color specified.
- Added styleOverrides in the theme to make the buttons look the same as before.
- *The ProjectTimeAndPlaceSection commit is accidentally. I remove it later in Cleanup step 2. This is a file from a different branch that wasn't properly stashed*

### [Step 8: Fix components: Outlined buttons, Grid, Slider, Divider, Checkbox, Chip; Theme import](https://github.com/climateconnect/climateconnect/pull/1223/commits/11140f157c749af6b5f996b1eb706bfcd7362cfa)

This commit includes a bunch of smaller changes to components to make sure they still work like before. This is the last commit in the PR that changes functionality.
- **Chip**: The new default color is primary which makes our currently grey chips green. I decided to overwrite the "secondary" color for Chips in theme and change it to the old Chip color. This ensures Chips look the same as before.
- **Checkbox**: In mui 5 the default color for Checkbox is now primary. Since all our Checkboxes had color="primary" anyways I removed all the explicit `color=primary` statements as they are now redundant.
- **Button**: Outlined buttons also switched to `color="primary"` by default which made the filter buttons green. I added a color="grey" option for outlined buttons in the theme (additional to the option for contained buttosn added in Step 7) and applied it to all outlined buttons. The style override for theme.ts was retrieved from the same code sandbox as in step 7.
- **Grid**: With the old Grid component margins in multiple grids were broken (e.g. recommended projects in the sidebar of the projects page). Therefore I decided to migrate to Grid v2 using [this migration guide](https://mui.com/material-ui/migration/migration-grid-v2/). This makes our grids more future proof and gives us more options. Some of the new options might be unstable but I'm not using any of them.
- **Divider** I used the css property `borderColor` instead of `backgroundColor` as described in the migration guide
- **Slider** I used `size="small"` and removed margins to make the IdeaRatingsSlider look the same as before. Without this change the slider was distorted.

### CODE REVIEW

File: frontend/pages/faq.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ const useStyles = makeStyles((theme) => {
   return {
     headerImageContainer: {
       marginBottom: theme.spacing(3),
-      [theme.breakpoints.down("md")]: {
+      [theme.breakpoints.down("lg")]: {
```
Comment (ddhanesha) [inline frontend/pages/faq.tsx:19]: Why did you change breakpoints from "md" to "lg"? There are other instances of this so wonder about the reasoning.

File: frontend/src/components/account/SettingsPage.tsx [hunk 2]
```diff
@@ -453,7 +453,6 @@ export default function SettingsPage({ settings, setSettings, token, setMessage
                 checked={emailPreferences[key]}
                 onChange={() => handlePreferenceChange(event, key)}
                 name={key}
-                color="primary"
```
Comment (ddhanesha) [inline frontend/src/components/account/SettingsPage.tsx:456]: Why did you remove color from this button? I see that in some files you added `color` parameter to the button. What is different here?

File: frontend/src/components/dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -32,7 +23,7 @@ const useStyles = makeStyles((theme) => {
       minWidth: 300,
       width: "100%",
       borderRadius: 5,
-      border: theme.borders.thick,
+      border: `3px solid ${theme.palette.primary.main}`,
```
Comment (ddhanesha) [inline frontend/src/components/dashboard/Dashboard.tsx:26]: Hmmmmm, did you determine this based on MUI's suggestion?

File: frontend/src/components/dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -54,7 +45,7 @@ const useStyles = makeStyles((theme) => {
       // TODO(design): what color should this actually be -- I
       // don't see it represented in the XD mockup? Ideally
       // it'd be from our emerging design system
-      border: theme.borders.thin,
+      border: `1px solid #e0e0e0`,
```
Comment (ddhanesha) [inline frontend/src/components/dashboard/Dashboard.tsx:48]: Same here, wonder how did you determine these changes?

File: frontend/src/components/feedback/FeedbackDialog.tsx [hunk 5]
```diff
@@ -70,14 +70,13 @@ export default function FeedbackDialog({ onClose, open, title, inputLabel, maxLe
...
             rows={4}
-            rowsMax={15}
+            maxRows={15}
             required
           />
           <Checkbox
             id={"feedbackcheckbox"}
             className={classes.checkbox}
             checked={checked}
-            color="primary"
```
Comment (ddhanesha) [inline frontend/src/components/feedback/FeedbackDialog.tsx:80]: Again, why did you remove the color from the checkbox?

File: frontend/src/components/general/BottomNavigation.tsx [hunk 6]
```diff
@@ -97,6 +104,7 @@ export default function BottomNavigation({
         {saveAsDraft && (
           <Button
             variant="contained"
+            color="grey"
```
Comment (ddhanesha) [inline frontend/src/components/general/BottomNavigation.tsx:107]: I see you added `color=grey` in multiple places. Is that because the style is not recognizing the color?

File: frontend/src/components/general/DatePicker.tsx [hunk 7]
```diff
@@ -25,8 +27,8 @@ export default function DatePicker({
     handleChange(selectedDate);
   };
   return (
-    <MuiPickersUtilsProvider utils={DateFnsUtils}>
-      <KeyboardDatePicker
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <DatePickerComponent
```
Comment (ddhanesha) [inline frontend/src/components/general/DatePicker.tsx:31]: Did you test this change locally? Is it working everywhere we use date picker?

File: frontend/src/components/hub/HubHeadlineContainer.tsx [hunk 8]
```diff
@@ -17,9 +18,9 @@ const useStyles = makeStyles((theme) => ({
 
     maxWidth: "800px",
     borderRadius: 5,
-    border: theme.borders.thick,
+    border: `3px solid ${theme.palette.primary.main}`,
```
Comment (ddhanesha) [inline frontend/src/components/hub/HubHeadlineContainer.tsx:21]: Again, curious how did you determine this?

File: frontend/src/components/ideas/IdeaRatingSlider.tsx [hunk 9]
```diff
@@ -62,7 +61,7 @@ export default function IdeaRatingSlider(props) {
   const classes = useStyles({ averageRating: props.averageRating });
   return (
     <div className={classes.wrapper}>
-      <Slider {...props} classes={classes} />
+      <Slider {...props} classes={classes} size="small" />
```
Comment (ddhanesha) [inline frontend/src/components/ideas/IdeaRatingSlider.tsx:64]: Is the styles not recognizing size for the slider?

File: frontend/src/components/layouts/layout.tsx [hunk 10]
```diff
@@ -14,7 +14,7 @@ import LayoutWrapper from "./LayoutWrapper";
 const useStyles = makeStyles((theme) => ({
   mainHeading: {
     textAlign: "center",
-    margin: `${theme.spacing(4)}px 0`,
+    margin: `${theme.spacing(4)} 0`,
```
Comment (ddhanesha) [inline frontend/src/components/layouts/layout.tsx:17]: is `px` automatically recognized in MUI5?

File: frontend/src/components/account/SettingsPage.tsx (hunk 2 above)
Comment (positiveimpact) [reply frontend/src/components/account/SettingsPage.tsx:456]: From the PR description: "With mui 5 the breakpoints have changed and for the same behavior each breakpoint needs to be upgraded by one (e.g. "xs" -> "sm"). I have made this change by running a codemod provided by mui."

File: frontend/src/components/dashboard/Dashboard.tsx (hunk 3 above)
Comment (positiveimpact) [reply frontend/src/components/dashboard/Dashboard.tsx:26]: While changing the theme I realized having a custom property "thick"/"thin" in the theme for just 2 files was a bit overkill. All I did here was basically directly putting the css into the "Dashboard" file and removing the generic property from the theme

File: frontend/src/components/dashboard/Dashboard.tsx (hunk 4 above)
Comment (positiveimpact) [reply frontend/src/components/dashboard/Dashboard.tsx:48]: Same here

File: frontend/src/components/feedback/FeedbackDialog.tsx (hunk 5 above)
Comment (positiveimpact) [reply frontend/src/components/feedback/FeedbackDialog.tsx:80]: From Pr description step 8: Checkbox: In mui 5 the default color for Checkbox is now primary. Since all our Checkboxes had color="primary" anyways I removed all the explicit color=primary statements as they are now redundant.

File: frontend/src/components/general/BottomNavigation.tsx (hunk 6 above)
Comment (positiveimpact) [reply frontend/src/components/general/BottomNavigation.tsx:107]: From PR description step 7: In mui 5 all buttons the default grey color has been removed and all buttons have color="primary" by default now. To keep the old colors I applied the changes from the [CodeSandbox demo](https://codesandbox.io/s/mimic-v4-button-default-color-bklx8?file=/src/Demo.tsx) provided in the migration guide. Following the guide I introduced a "grey" color to the button component and applied it on all contained buttons that didn't have a color specified.

File: frontend/src/components/general/DatePicker.tsx (hunk 7 above)
Comment (positiveimpact) [reply frontend/src/components/general/DatePicker.tsx:31]: Yes

File: frontend/src/components/hub/HubHeadlineContainer.tsx (hunk 8 above)
Comment (positiveimpact) [reply frontend/src/components/hub/HubHeadlineContainer.tsx:21]: Same as above, just directly using the css code rather than calling it from the theme

File: frontend/src/components/ideas/IdeaRatingSlider.tsx (hunk 9 above)
Comment (positiveimpact) [reply frontend/src/components/ideas/IdeaRatingSlider.tsx:64]: From PR description step 8: Slider I used size="small" and removed margins to make the IdeaRatingsSlider look the same as before. Without this change the slider was distorted.

File: frontend/src/components/layouts/layout.tsx (hunk 10 above)
Comment (positiveimpact) [reply frontend/src/components/layouts/layout.tsx:17]: Yes

**Comment by positiveimpact** [comment]
Thanks for the review @ddhanesha. Going to merge this :)
